src/components/app/MenuButtons/MetaOverheadStatistics.js

```diff
@@ -0,0 +1,161 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import * as React from 'react';
+import {
+  formatNanoseconds,
+  formatPercent,
+} from '../../../utils/format-numbers';
+
+import type { ProfilerOverhead } from '../../../types/profile';
+
+import './MetaOverheadStatistics.css';
+
+// Profiler overhead statistics keys that have max/min/mean values.
+type StatKeys = 'Overhead' | 'Cleaning' | 'Counter' | 'Interval' | 'Lockings';
+
+type Props = {
+  profilerOverhead?: ProfilerOverhead[],
+};
+
+/**
+ * This component formats the profile's meta information into a dropdown panel.
+ */
+export class MetaOverheadStatistics extends React.PureComponent<Props> {
+  render() {
+    const { profilerOverhead } = this.props;
+
+    const calculatedStats: Map<StatKeys, ProfilerStats> = new Map();
+    // These 3 values only have single values, so we are not using ProfilerStats class for them.
+    // Weightedly accumulated value of `overheadDurations. It's used to calculate mean value.
+    let overheadDurations = 0;
+    // Weightedly accumulated value of `overheadPercentage. It's used to calculate mean value.
+    let overheadPercentage = 0;
+    // Weightedly accumulated value of `profiledDuration. It's used to calculate mean value.
+    let profiledDuration = 0;
+    // Total sampling count is used to calculate mean values of the 3
+    // statistics values above.
+    let totalSamplingCount = 0;
+
+    // Older profiles(before FF 70) don't have any overhead info. Don't show anything if
+    // that's the case.
+    if (profilerOverhead) {
+      // Overhead keys that have min/max/mean values to loop.
+      const statKeys = [
+        'Overhead',
+        'Cleaning',
+        'Counter',
+        'Interval',
+        'Lockings',
+      ];
+
+      for (const overhead of profilerOverhead) {
+        const { statistics } = overhead;
+        const { samplingCount } = statistics;
+
+        // Calculation the single values without any loop, it's not worth it.
+        overheadDurations += statistics.overheadDurations * samplingCount;
+        overheadPercentage += statistics.overheadPercentage * samplingCount;
+        profiledDuration += statistics.profiledDuration * samplingCount;
+        totalSamplingCount += samplingCount;
```

Comment:
I feel like we could do these calculations using `ProfilerStats`, maybe it could be made more generic. But it's not a big deal.